Sobel/sobel.c

```diff
@@ -0,0 +1,130 @@
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <pthread.h>
+
+pthread_mutex_t mtx;
+
+char** src;
+char** dst;
+int width, height;
+
+struct str {
+    int start;
+    int end;
+};
+
+int xGradient(int x, int y) {
+    return src[y - 1][x - 1] +
+           2 * src[y][x - 1] +
+           src[y + 1][x - 1] -
+           src[y - 1][x + 1] -
+           2 * src[y][x + 1] -
+           src[y + 1][x + 1];
+}
+ 
+ 
+int yGradient(int x, int y) {
+    return src[y - 1][x - 1] +
+           2 * src[y - 1][x] +
+           src[y - 1][x + 1] -
+           src[y + 1][x - 1] -
+           2 * src[y + 1][x] -
+           src[y + 1][x + 1];
+}
+
+void* SobelForLines(void* args) {
+    int start = ((struct str*)args)->start;
+    int end = ((struct str*)args)->end;
+    for (int y = start; y < end; y++) {
+        for (int x = 1; x < width - 1; x++) {
+            int gx = xGradient(x, y);
+            int gy = yGradient(x, y);
+            int sum = abs(gx) + abs(gy);
+            
+            pthread_mutex_lock(&mtx);
+            dst[y][x] = (char)(sum > 255 ? 255 : (sum < 0 ? 0 : sum));
+            pthread_mutex_unlock(&mtx);
+        }
+    }
+}
+
+
+int main() {
+    FILE *input, *output;
+    if ((input = fopen("image.bmp", "r")) == NULL) {
+        return -1;
+    }
+    
+    char* info = (char*)malloc(54);
```

Comment:
In C there is no need to explicit cast for malloc, it returns void *, which automatically converted to correct type. It relates to all variables in this PR.
```suggestion
    char* info = malloc(54);
```